### PR TITLE
Create an API-ready case interface in BulkCaseForm to facilitate forthcoming validation

### PR DESCRIPTION
**Pull request by axmb**
Primarily need to define a type in the component to hold case data such that we can cleanly construct each case once, and reuse the object for validation and eventual submission. Cutting this as a separate PR so that the eventual surfacing of validation errors isn't too massive/dense.

See the note on the partial type I've created. I'm hoping that we eventually find cycles to label fields optional in `Case` if they're optional in the API case document.

### CODE REVIEW

**Comment by axmb** [comment]
[Surface detailed stats and errors for bulk upload](https://trello.com/c/xiVeNvnn/280-surface-detailed-stats-and-errors-for-bulk-upload)

File: verification/curator-service/ui/src/components/BulkCaseForm.tsx [hunk 1]
```diff
@@ -79,14 +79,27 @@ interface ParsedCase {
...
+    latitude?: number;
+    longitude?: number;
     locationName?: string;
 
     // Bulk upload specific data
     caseCount?: number;
 }
 
+type RecursivePartial<T> = {
+    [P in keyof T]?: RecursivePartial<T[P]>;
```
Comment (allysonjp715) [inline verification/curator-service/ui/src/components/BulkCaseForm.tsx:91]: 🤔  This is pretty confusing, could you add a comment to the type describing it's use?

File: verification/curator-service/ui/src/components/Case.tsx [hunk 2]
```diff
@@ -45,6 +45,7 @@ export interface Location {
     geometry: Geometry;
     name: string;
     place: string;
+    query?: string;
```
Comment (allysonjp715) [inline verification/curator-service/ui/src/components/Case.tsx:48]: Is this needed? The type description says `// Case definitions as returned by the /api/cases endpoint.` and I don't think query is ever returned from the backend, is it? If it's needed perhaps change the type description.

Comment (axmb) [reply verification/curator-service/ui/src/components/Case.tsx:48]: Good call out, I'll change the type description. This is required to POST cases without a prior geocode. In the new case form, we use the geocode `suggest` endpoint, which takes care of completing the `location` object prior to use POSTing the new case. But for non-human-interactive usage, we're currently requiring a `query` field in `location` [in order to geocode](https://github.com/open-covid-data/healthmap-gdo-temp/blob/master/verification/curator-service/api/src/controllers/cases.ts#L124).

File: verification/curator-service/ui/src/components/BulkCaseForm.tsx (hunk 1 above)
Comment (axmb) [reply verification/curator-service/ui/src/components/BulkCaseForm.tsx:91]: Yeah, sorry about that. Done.

**Pull request merged**